scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerStateRecoverHelper.java

```diff
@@ -62,31 +70,26 @@ public RecoveredSchedulerState recover(long loadJobPeriod, RMProxy rmProxy) {
         Vector<InternalJob> pendingJobs = new Vector<>();
         Vector<InternalJob> runningJobs = new Vector<>();
 
+        ExecutorService recoverRunningTasksThreadPool = Executors.newFixedThreadPool(PASchedulerProperties.SCHEDULER_PARALLEL_SCHEDULER_STATE_RECOVER_NBTHREAD.getValueAsInt());
+
         for (InternalJob job : notFinishedJobs) {
-            switch (job.getStatus()) {
-                case PENDING:
-                    pendingJobs.add(job);
-                    break;
-                case STALLED:
-                case RUNNING:
-                case IN_ERROR:
-                    runningJobs.add(job);
-                    recoverRunningTasksOrResetToPending(job, job.getITasks(), rmProxy);
-                    break;
-                case PAUSED:
-                    if ((job.getNumberOfPendingTasks() + job.getNumberOfRunningTasks() +
-                         job.getNumberOfFinishedTasks()) == 0) {
-                        pendingJobs.add(job);
-                    } else {
-                        runningJobs.add(job);
-                        recoverRunningTasksOrResetToPending(job, job.getITasks(), rmProxy);
-                    }
-                    break;
-                default:
-                    throw new IllegalStateException("Unexpected job status: " + job.getStatus());
-            }
+            recoverJob(rmProxy, pendingJobs, runningJobs, job, recoverRunningTasksThreadPool);
+        }
+
+        recoverRunningTasksThreadPool.shutdown();
+
+        boolean terminatedWithoutTimeout;
+
+        try {
+            terminatedWithoutTimeout = recoverRunningTasksThreadPool.awaitTermination(PASchedulerProperties.SCHEDULER_PARALLEL_SCHEDULER_STATE_RECOVER_TIMEOUT.getValueAsInt(),
+                                                                                      TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)

Reply:
Can you use Thread.currentThread.interrupt() ?

Reply:
Done.